### ISSUE TITLE
Spotlight: remotely ungrouping result group bug

### ISSUE THREAD

**Issue by nickoferrall** [ParabolInc/parabol#5627]
A reflection doesn't return to the correct position in the Spotlight modal if it's remotely ungrouped in the results section.

To reproduce:
- Alice opens the Spotlight in client A. One of the result groups contains 2 reflections
- Bob, in client B, ungroups the result group that contains 2 reflections
- Alice sees that the reflection returns to the incorrect position in the result groups

### AC

- [ ] If a results reflection group is remotely ungrouped, it returns to the expected position in the Spotlight

**Comment by nickoferrall** [comment]
This is caused by the`isFinishingRemoteDragging` check [here](https://github.com/ParabolInc/parabol/blob/2360f8cb7609ba483b2473c9e7635c204cdf7a85/packages/client/components/ReflectionGroup/DraggableReflectionCard.tsx#L143) which was added to fix https://github.com/ParabolInc/parabol/issues/5415

We need to find a way of preventing remote reflections from losing their position when the Spotlight is opened while also making sure that result groups remain visible when they're remotely ungrouped.

**Comment by nickoferrall** [comment]
Hey @BartoszJarocki, if you're looking for a new issue and you'd like to return to the painful world of the `DraggableReflectionCard` ref logic, this could be a good one 😅  No worries if not!

**Comment by BartoszJarocki** [comment]
@nickoferrall challenge accepted ;) Seems like https://github.com/ParabolInc/parabol/issues/5415 won't let me forget about it too soon 😂

**Comment by nickoferrall** [comment]
Hey @BartoszJarocki, I found a solution for this that's not related to the `isFinishingRemoteDragging` logic, but the `isDroppable` logic in `reflectionGroup` instead! I'll push an update, most likely on Monday, and tag this issue in the PR.

**Comment by nickoferrall** [comment]
Spoke too soon - this is still related to the `isFinishingRemoteDragging` logic 😅

**Comment by BartoszJarocki** [comment]
No problem, I'll most probably have a time on Monday to start the investigation 😅 Please let me know if anything changes! 🙏

**Comment by nickoferrall** [comment]
Will do!

**Comment by BartoszJarocki** [comment]
@nickoferrall After updating from the newest master the behavior has changed. Now the remote ungroup can be seen in the realtime but the card returns to the wrong position. Should I continue working on this or do we have a separate issue for this behavior? 
https://user-images.githubusercontent.com/1017620/142815587-9827d2fb-6260-4d34-a094-8b6160bbf78c.mp4

**Comment by nickoferrall** [comment]
@BartoszJarocki it would be great if you could continue working on this as it looks like the reason the remote ungrouping positioning is off is because of the `isFinishingRemoteDragging` return statement. 

In the gif below, we see the current behaviour, then I comment out the `isFinishingRemoteDragging` return statement, and then we see the desired behaviour:

![remote-ungroup-position](https://user-images.githubusercontent.com/39854876/142856843-d5dea6bd-1730-4f41-9402-66a1f36b494d.gif)

I don't mind if we create a new issue or update the AC here. I think the title of the issue is still accurate so I'll update the AC, but free to close & create a new one if you prefer.

**Comment by BartoszJarocki** [comment]
Ok cool, no need for a new issue!

**Comment by BartoszJarocki** [comment]
@nickoferrall unfortunately, I have to ask for help. I'm feeling stuck and I'm running out of ideas. 

Technically, the solution for both problems (the initial one, and the one from this task) should be something like that (pseudo code)
```
  // this will run only once, in case spolitht was closed, now it's opened and the remote drag is in progress
  const [isFinishingRemoteDragging, setIsFinishingRemoteDragging] = useState(
    () => isDropping && hasSpotlightJustOpened && !!remoteDrag
  )
  // if the card was finishing remote drag and it was dropped into the original position, let it behave normally
  useEffect(() => {
    if (isFinishingRemoteDragging && !isDropping) {
      setIsFinishingRemoteDragging(false)
    }
  }, [isFinishingRemoteDragging, isDropping])
```
The key for it to work is to be able to know how to calculate `hasSpotlightJustOpened`, ie. the spotlight wasn't opened but it's open now, remote drag is in progress so let it finish and then, do not touch anything. 

With the current solution (code below), I'm using `isSpotlightOpen` as I relied on the state initializer function being called once for a given card. It's not true, either components are being recreated somewhere, or React for some reason calls this `() => isDropping && isSpotlightOpen && !!remoteDrag` function at least a couple of times. As a result, even if the spotlight is already opened and there's remote drag starting, it'll be called and `isFinishingRemoteDragging` will be set to true, which will cause this exact bug. 
```
  // if spotlight was just opened and card is in the middle of dropping we let it drop into original position
  const [isFinishingRemoteDragging, setIsFinishingRemoteDragging] = useState(
    () => isDropping && isSpotlightOpen && !!remoteDrag
  )
  // if the card was finishing remote drag and it was dropped into the original position, let it behave normally
  useEffect(() => {
    if (isFinishingRemoteDragging && !isDropping) {
      setIsFinishingRemoteDragging(false)
    }
  }, [isFinishingRemoteDragging, isDropping])
```
I'm not sure how to proceed further with it. I'd appreciate it if you could take a look. I'm out starting tomorrow and I'll be back on Monday, then I can get back to it. I do feel really bad because I have already spent a lot of time on it.

**Comment by nickoferrall** [comment]
Ah all good, @BartoszJarocki! Thanks for sharing your progress so far, and no problem at all. The drag ref callback edge cases are extremely painful! I'd be happy to dig into this later today and see if I can make any headway. I'll keep you posted 👍

**Comment by BartoszJarocki** [comment]
That's great, thank you! If you won't find anything maybe we should pair on Monday and see if we can figure it out together?

**Comment by nickoferrall** [comment]
That sounds great! If I'm still battling away on Monday, we can cry together on Zoom 👌

**Comment by BartoszJarocki** [comment]
Perfect, thank you! 🤞
